ruma-events-macros/src/event_enum.rs

```diff
@@ -1,28 +1,47 @@
 //! Implementation of event enum and event content enum macros.
 
-use proc_macro2::TokenStream;
+use proc_macro2::{Span, TokenStream};
 use quote::quote;
 use syn::{
     parse::{self, Parse, ParseStream},
     Attribute, Expr, ExprLit, Ident, Lit, LitStr, Token,
 };
 
+use crate::event_names::{
+    ANY_BASIC_EVENT, ANY_EPHEMERAL_EVENT, ANY_MESSAGE_EVENT, ANY_STATE_EVENT,
+    ANY_SYNC_EPHEMERAL_EVENT, ANY_SYNC_MESSAGE_EVENT, ANY_SYNC_STATE_EVENT, ANY_TO_DEVICE_EVENT,
+};
+
+type HasField = fn(&Ident) -> bool;
+
+/// This const is used to generate the accessor methods for the `Any*Event` enums.
+///
+/// DO NOT alter the field names unless the structs in `ruma_events::event_kinds` changes.
+const EVENT_FIELDS: &[(&str, HasField)] = &[
+    ("origin_server_ts", has_room_event_fields),
```

Comment:
Maybe instead of functions, we could just have `&[EventKindA, EventKindB, EventKindC]`? Since all the functions do is the equivalent of `event_kind_list.contains(id)`, inlining them here as just the lists (and rewriting the ones that are not that). Or are they used in other places too?

Reply:
Is that what you were thinking? I think that works well!

To bad about the `&ident.to_string().as_str()` part, not much that could be done since it's `const`.